src/main/java/com/project/fintech/service/AccountService.java

```diff
@@ -0,0 +1,146 @@
+import com.project.fintech.persistence.repository.ArchivedTransactionRepository;
+import com.project.fintech.persistence.repository.UserRepository;
+import java.security.SecureRandom;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class AccountService {
+
+    private final Long MAX_ACCOUNT_COUNT = 3L;
+    private final String SERVICE_CODE = "177";
+
+    private final AccountRepository accountRepository;
+    private final TransactionMapper transactionMapper;
+    private final UserRepository userRepository;
+    private final ArchivedTransactionRepository archivedTransactionRepository;
+
+    /**
+     * email 로 User 가져오기
+     * @param email
+     * @return User
+     */
+    @Transactional
+    public User getUserByEmail(String email) {
+        return userRepository.findByEmail(email)
+            .orElseThrow(() -> new CustomException(ErrorCode.USER_NOT_FOUND));
+    }
+
+    /**
+     * account id로 Account 가져오기
+     *
+     * @param accountId
+     * @return Account
+     */
+    @Transactional
+    public Account getAccountById(Long accountId) {
+        return accountRepository.findById(accountId)
+            .orElseThrow(() -> new CustomException(ErrorCode.ACCOUNT_NOT_FOUND));
+    }
+
+    /**
+     * 계좌 생성하기
+     *
+     * @param user
+     * @return created account
+     */
+    @Transactional
+    public Account createAccount(User user) {
+        long accountCounts = accountRepository.countByUser(user);
+        if (accountCounts >= MAX_ACCOUNT_COUNT) {
+            throw new CustomException(ErrorCode.ACCOUNT_CREATION_LIMIT_EXCEEDED);
+        }
+        Account account = Account.builder().accountNumber(createAccountNumber()).user(user).build();
+        user.getAccount().add(account);
+        return account;
+    }
+
+    /**
+     * 게좌 삭제하기(Soft delete)
+     *
+     * @param user
+     * @param account
+     */
+    @Transactional
+    public void deleteAccount(User user, Account account) {
+        if (user.getAccount().stream().noneMatch(userAccount -> userAccount == account)) {
+            throw new CustomException(ErrorCode.ACCOUNT_USER_MISMATCH);
+        }
+        account.setStatusDisable();
+        moveTransactionsToArchive(account);
+    }
+
+    /**
+     * 해당 계좌번호의 transaction을 Archived transaction 테이블로 옮기기
+     *
+     * @param account
+     */
+    private void moveTransactionsToArchive(Account account) {
+        List<Transaction> transactionsList = account.getTransactions();
+
+        if (!transactionsList.isEmpty()) {
+            List<ArchivedTransaction> archivedTransactionList = transactionsList.stream()
+                .map(transactionMapper::toArchivedTransactions).toList();
+            archivedTransactionRepository.saveAll(archivedTransactionList);
+            account.getTransactions().clear();
+        }
```

Comment:
빠른 return 하는 방식에 대해 적용해보면 좋겠네요. 언제는 빠른 return 을 쓰는게 유리하고, 언제는 if 중첩을 사용하는게 유리할지 고민해보면 좋겠어요.
```suggestion
        if (transactionsList.isEmpty()) {
            return
        }
        
        List<ArchivedTransaction> archivedTransactionList = transactionsList.stream()
            .map(transactionMapper::toArchivedTransactions).toList();
        archivedTransactionRepository.saveAll(archivedTransactionList);
        account.getTransactions().clear();
```